### ISSUE TITLE
`carthage update` does not build framework

### ISSUE THREAD

**Issue by justinmakaila** [martijnwalraven/meteor-ios#25]
Output:
```
*** Cloning meteor-ios
*** Checking out meteor-ios at "0.1.2"
*** xcodebuild output can be found in /var/folders/pj/0_h5pbg1787dcz5j24_rs36m0000gn/T/carthage-xcodebuild.JQWjmo.log
Project "Pods.xcodeproj" has no shared schemes
```

**Comment by thatjuan** [comment]
same issue here

**Comment by justinmakaila** [comment]
Digging through, I see that `Pods.xcodeproj` does in fact not have any shared schemes, but I'm confused  as to why `carthage` is even looking in it. Probably has something to do with it automatically searching all `.xcodeproj`'s for shared schemes

**Comment by thatjuan** [comment]
I also cloned the repo, opened the workspace, but I'm unable to build any of the targets :/

**Comment by justinmakaila** [comment]
Adding an explicit commit to my `Cartfile.resolved` solved the problem.

Cartfile.resolved:
```
github "martijnwalraven/meteor-ios" "3b598cc9421df9affbef155ee3ce6dc5ff0572a9"
```
I think we need to bump the version to reflect the latest commit.

**Comment by thatjuan** [comment]
@justinmakaila cool, that worked for me too.. So this might be a Carthage bug?

**Comment by thatjuan** [comment]
@justinmakaila after installing it, did you run into this by any chance?

https://github.com/martijnwalraven/meteor-ios/issues/26

**Comment by justinmakaila** [comment]
I'm not sure, I'm still writing code to test.

It's probably an issue because the dependencies are not specified within a `Cartfile.resolved` in the repository.

I'll take a look in a bit, see if I can file a PR.

**Comment by martijnwalraven** [comment]
I've released a new version that should at least allow the Meteor framework to build again using Carthage. But I'm afraid that isn't of much use because of #26.

Having to support both CocoaPods and Carthage is a pain, so suggestions for improvements are very welcome. I'm struggling with finding a way to embed dependencies specified as Pods, but maybe that is not the way to go.

Using `Cartfile.resolved` won't work for dependencies that do not have an Xcode project but only build using CocoaPods (such as https://github.com/mattt/InflectorKit).

**Comment by thatjuan** [comment]
@martijnwalraven thank you for looking into this. 

I was thinking about forking InflectorKit and turning it into a framework that can be installed by Carthage, but I don't have a lot of time so I ended up switching to CocoaPods.

**Comment by justinmakaila** [comment]
@thatjuan and @martijnwalraven I have a fork of InflectorKit, PocketSocket, and SimpleKeychain which solves this issue, I just need to get an issue I'm having personally with Carthage worked out.

I've never found CocoaPods to be a desirable solution, as the workflow requires writing more code, where Carthage is as simple as changing some Xcode settings (once!) and using `git tag` to maintain versions.

That said, both are almost required to be considered a "modern" cocoa framework, which is a bummer when you have dependencies which are not "modern".

**Comment by thatjuan** [comment]
@justinmakaila we're on the same page regarding CocoaPods. I switched back only because I really needed to get meteor-ios working for this project. 

Thanks for working on getting this up and running with Carthage and let me know if there's anything I can pitch in. I've run into several issues with Carthage as well, but I'm sure it'll mature fairly quickly.

**Comment by justinmakaila** [comment]
Forks are here:

[PocketSocket](http://github.com/justinmakaila/pocketsocket)
[InflectorKit](http://github.com/justinmakaila/inflectorkit)
[SimpleKeychain](http://github.com/justinmakaila/simplekeychain)

I also have a fork of this repo, which has a Cartfile which points to those dependencies [here](http://github.com/justinmakaila/meteor-ios).

Once my issues with Carthage are resolved, I'll file a PR.

**Comment by martijnwalraven** [comment]
Great work! Ideally, I'd like to find a way to build the Meteor framework target using the Cartfile dependencies. Right now, the target uses the Pods project generated by CocoaPods, which means there could be version conflicts with the versions specified in the Cartfile. 

Do you think it would make sense to include the dependent projects in the workspace and link to the framework targets? Maybe using submodules would help here? Or maybe include the Carthage/Build directory? In any case, it would be nice if we could get rid of the checked in Pods directory completely. I just want to make sure people cloning the repository can run the examples without using either CocoaPods or Carthage.

**Comment by martijnwalraven** [comment]
That looks good! I merged your pull request. The test targets weren't working, so I had to make a few changes. Some of these had to do with directories I had renamed in a previous commit.

But I also had to add OCMock.framework to Carthage/Build. OCMock doesn't support Carthage unfortunately, but I did find a fork which at least has an iOS framework target. I added it to Cartfile.private but it still wouldn't build. For now, I've built it manually and copied the .framework file to the build directory.

I also had to add a 'Copy Dependencies' phase to the test targets. I'm not sure why it wouldn't use the embedded frameworks in Meteor.framework. Do you know anything about this?

**Comment by justinmakaila** [comment]
This should also be closed with #27.

I cleaned up some of the build phases of the targets, the "Copy Dependencies" on the Meteor target was overkill, because those dependencies are linked, and shouldn't be copied into the actual `Meteor.framework`.

I checked OCMock into the repo and got all the Unit Tests to pass, but I'm running into a lot of failures with the server integration tests. Do those pass for you?

**Comment by martijnwalraven** [comment]
The server integration tests require a running test server (run 'meteor' in 'Tests/Test Server'). There are two tests that fail for me right now, both having to do with logging in. I think that might have to do with the tests expecting a user with a certain password to exist in the test server. I'll look into it.
